### PR TITLE
Split out integration test plumbing (#2594) (#2300)

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Part of #2594
Part of #2300

# Rationale for this change
 

The way the integration test plumbing is muddled with the core logic is incredibly confusing, and complicates splitting apart the crates (#2594) and adding features flags (#2300)

# What changes are included in this PR?

This moves the integration test plumbing to the existing integration-testing. I originally moved it to a separate crate, but this seemed redundant

# Are there any user-facing changes?

Technically this functionality was being published, it now isn't. I have confirmed DataFusion isn't making use of this.

### CODE REVIEW

File: arrow/src/ipc/writer.rs [hunk 1]
```diff
@@ -1270,47 +1270,43 @@ mod tests {
...
         let values: Vec<Option<i32>> = vec![];
         let array = Int32Array::from(values);
         let record_batch =
             RecordBatch::try_new(Arc::new(schema.clone()), vec![Arc::new(array)])
                 .unwrap();
+
+        let mut file = tempfile::tempfile().unwrap();
+
         {
-            let file =
```
Comment (tustvold) [inline arrow/src/ipc/writer.rs:1295]: This is a drive-by cleanup

**Review by liukun4515 — APPROVED** [review]
